.github/workflows/end2end.yaml

```diff
@@ -0,0 +1,406 @@
+          SCALITY_OCI_RW_USER: ${{ secrets.SCALITY_OCI_RW_USER }}
+          SCALITY_OCI_RW_PASSWORD: ${{ secrets.SCALITY_OCI_RW_PASSWORD }}
+      - name: Configure GIT
+        run: git config --global url."https://${GIT_ACCESS_TOKEN}@github.com/".insteadOf "https://github.com/"
+        env:
+          GIT_ACCESS_TOKEN: ${{ secrets.GIT_ACCESS_TOKEN }}
+      - name: Build ISO
+        run: PATH="/usr/local/bin:$PATH" bash -x ./build.sh
+        working-directory: ./solution
+      - name: Build base ISO
+        run: PATH="/usr/local/bin:$PATH" bash -x ./build.sh
+        working-directory: ./solution-base
+      - name: Upload ISO
+        uses: scality/action-artifacts@v3
+        with:
+          method: upload
+          url: https://artifacts.scality.net
+          user: ${{ secrets.ARTIFACTS_USER }}
+          password: ${{ secrets.ARTIFACTS_PASSWORD }}
+          source: "solution/_build"
+      - name: Upload base ISO
+        uses: scality/action-artifacts@v3
+        with:
+          method: upload
+          url: https://artifacts.scality.net
+          user: ${{ secrets.ARTIFACTS_USER }}
+          password: ${{ secrets.ARTIFACTS_PASSWORD }}
+          source: "solution-base/_build"
+
+  build-test-image:
+    runs-on:
+      - self-hosted
+      - centos7
+      - large
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to Registry
+        uses: docker/login-action@v1.10.0
+        with:
+          username: "${{ secrets.REGISTRY_LOGIN }}"
+          password: "${{ secrets.REGISTRY_PASSWORD }}"
+          registry: registry.scality.com
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: buildx-zenko-${{ github.sha }}
+          restore-keys: |
+            buildx-zenko
+      - name: Generate end2end config yaml
+        run: |-
+          cd tests/zenko_tests
+          if [[ "${ENABLE_RING_TESTS}" == "false" ]]; then
+            envsubst < 'e2e-config.yaml.template' > 'e2e-config.yaml'
+          else
+            envsubst < 'ring-e2e-config.yaml.template' > 'e2e-config.yaml'
+          fi
+          cat e2e-config.yaml
+          echo 'Generated e2e-config.yaml file'
+      - name: Build and push CI image
+        uses: docker/build-push-action@v3.1.0
+        with:
+          push: true
+          context: ./tests/zenko_tests
+          tags: "${{ env.E2E_IMAGE_NAME }}:${{ env.E2E_IMAGE_TAG }}"
+          cache-from: type=local,src=/tmp/.buildx-cache
+          cache-to: type=local,mode=max,dest=/tmp/.buildx-cache-new
+      - name: Move cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+  
+  end2end-http:
+    needs: [build-test-image, check-dashboard-versions]
+    runs-on:
+      - self-hosted
+      - ubuntu
+      - focal
+      - xlarge
```

Comment:
Keep an eye on a blog post I'll post in a few. If you remove the `self-hosted` tag you're in for a treat.

Reply:
This looks amazing! So I just removed the `self-hosted` line (visual studio is saying we have a syntax error but I guess it's because I'm not up to date or it is still a beta).
Is it enough to be able to use these new runners? I still see gcloud & openstack in the builds.

For the backbeat tests, having a more powerful environment will greatly help (for the other, the current CPU is enough).

Reply:
Should be enough, I'll have a look and let you know

Reply:
I got a build on the new runner! https://github.com/scality/Zenko/actions/runs/3066374814/jobs/4951502412#step:1:2

Reply:
Uhul!! 🎉 

My theory was, as the self-hosted are pre-created and pre attached, if we're not under a lot of load. GitHub actions will pick those as they are there and available. but if we reach a point where they are all busy, or being created, it will trigger a GitHub Hosted one.

I have nothing to back this up, but it feels right 😅 🤷